lib/geo.js

```diff
@@ -10,23 +10,31 @@ var assert = require('assert');
  */
 
 exports.nearFilter = function nearFilter(where) {
+  var keyPath = [];
   var result = false;
 
-  if (where && typeof where === 'object') {
-    Object.keys(where).forEach(function(key) {
-      var ex = where[key];
-
-      if (ex && ex.near) {
-        result = {
-          near: ex.near,
-          maxDistance: ex.maxDistance,
-          unit: ex.unit,
-          key: key,
-        };
-      }
-    });
+  searchForNear(where);
+
+  function searchForNear(node) {
+    if (node && typeof node === 'object') {
+      Object.keys(node).forEach(function(key) {
+        var ex = node[key];
+        keyPath.push(key);
+
+        if (ex && ex.near) {
+          result = {
+            near: ex.near,
+            maxDistance: ex.maxDistance,
+            unit: ex.unit,
+            key: keyPath.slice(0),
```

Comment:
@etsuo It LGTM 👍 
But I have same question as @Amir-61 mentioned, what does `key` mean and why do we keep track of the keys in query?

@bajtos @raymondfeng @ritch @superkhau Could you help confirm? Thanks!

And the other thought is I am not sure is this the best way to handle recursive query, another way to handle it is checking operator `and` and analyze the subfilters recursively, like what we do in [`setScopeValuesFromWhere`](https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/utils.js#L50)
But anyway your implementation also works fine to me 👍